fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -367,7 +367,7 @@ protected <M extends Message> CompletableFuture<Void> updateOneKeyAsync(@Nonnull
                 state.store.countKeyValue(FDBStoreTimer.Counts.DELETE_INDEX_KEY, FDBStoreTimer.Counts.DELETE_INDEX_KEY_BYTES, FDBStoreTimer.Counts.DELETE_INDEX_VALUE_BYTES,
                         keyBytes, valueBytes);
             }
-            if (state.store.isIndexWriteOnly(state.index) && state.index.isUnique()) {
+            if (isWriteOnlyOrUniquePending() && state.index.isUnique()) {
                 return removeUniquenessViolationsAsync(valueKey, savedRecord.getPrimaryKey());
```

Comment:
If successfully removed uniqueness violation entry (i.e. the counter is reduced to 1 and the entry was deleted), would It be acceptable to attempt converting the index to `READABLE`?